File: android/mqtt/src/main/java/com/mqtt/core/model/MqttConnectionState.kt
package com.mqtt.core.model

/**
 * Represents the current state of the MQTT connection
 */
sealed class MqttConnectionState {
    /**
     * Not connected and not attempting to connect
     */
    data object Disconnected : MqttConnectionState()

    /**
     * Attempting to establish connection
     */
    data object Connecting : MqttConnectionState()

    /**
     * Successfully connected to broker
     */
    data object Connected : MqttConnectionState()

    /**
     * Connection lost, attempting to reconnect
     * @param attempt Current reconnection attempt number
     */
    data class Reconnecting(val attempt: Int) : MqttConnectionState()

    /**
     * Connection failed with error
     * @param error The error that caused the failure
     */
    data class Error(val error: Throwable) : MqttConnectionState()

    /**
     * Waiting for network connectivity
     */
    data object WaitingForNetwork : MqttConnectionState()
}

File: iOS/MQTT Demo/MQTT Demo/Models/MQTTConnectionState.swift
//
//  MQTTConnectionState.swift
//  MQTT Demo
//
//  Created by Roman Tolmachev on 20-11-25.
//

import Foundation

/// Represents the current state of MQTT connection
enum MQTTConnectionState: Equatable {
    case disconnected
    case connecting
    case connected
    case disconnecting
    case error(String)
    
    var displayText: String {
        switch self {
        case .disconnected:
            return "Disconnected"
        case .connecting:
            return "Connecting..."
        case .connected:
            return "Connected"
        case .disconnecting:
            return "Disconnecting..."
        case .error(let message):
            return "Error: \(message)"
        }
    }
    
    var isConnected: Bool {
        if case .connected = self {
            return true
        }
        return false
    }
}
